sys/include/net/gnrc/ipv6/nc.h

```diff
@@ -118,7 +119,9 @@ extern "C" {
  *          </a>.
  */
 typedef struct {
+#ifdef MODULE_GNRC_NDP_NODE
```

Comment:
If you *only* build for 6LoWPAN, yes. For border routers with ethernet support or similar, this might be included regardless and multi-NDP support kicks in.